lmfit/model.py

```diff
@@ -455,25 +456,43 @@ def _parse_params(self):
         """Build parameters from function arguments."""
         if self.func is None:
             return
+        # need to fetch the following from the function signature:
+        #   pos_args: list of positional argument names
+        #   kw_args: dict of keyword arguments with default values
+        #   keywords_:  name of **kws argument or None
+        # 1. limited support for asteval functions as the model functions:
         if hasattr(self.func, 'argnames') and hasattr(self.func, 'kwargs'):
             pos_args = self.func.argnames[:]
+            keywords_ = None
             kw_args = {}
             for name, defval in self.func.kwargs:
                 kw_args[name] = defval
-            keywords_ = list(kw_args.keys())
+        # 2. modern, best-practice approach: use inspect.signature
+        elif sys.version_info.major == 3 and sys.version_info.minor > 4:
```

Comment:
can we do ```elif sys.version_info > (3, 4):``` here instead? I suppose we only use this for version above Python 3.5 because there it introduced the ```follow_wrapped``` parameter?

Reply:
yes, I did not test that it fails with 3.4, but the docs suggest it might.  We could just check major version and then be willing to fail for very old Python 3 versions, but I think ">3.4" is safe -- unless you disagree.  This does sort of mean that `lmfit` requires Py > 3.4, but I think we weren't promising to support 3.4 anyway.    We could just check major version and then be willing to fail for very old Python 3 versions, but I think ">3.4" is safe -- unless you disagree.

And, I'll fix the order of the imports!

Reply:
we already don't test 3.4 anymore and say in the docs: "Lmfit works with Python versions 2.7 and 3.5, 3.6, or 3.7." So I agree with checking here for > 3.4